nuitka/build/inline_copy/lib/scons-3.1.2/SCons/Platform/win32.py

```diff
@@ -199,9 +201,9 @@ def piped_spawn(sh, escape, cmd, args, env, stdout, stderr):
 
         # redirect output of non-redirected streams to our tempfiles
         if stdoutRedirected == 0:
-            args.append(">" + str(tmpFileStdout))
+            args.append(">" + str(tmpFileStdoutName))
```

Comment:
The actual execution would have to use these filenames, and ideally not use files at all, but just pipes I guess.

Reply:
Let's say we have `tmpFileStdoutName = "/tmp/random"`

And previously, the `mktemp` method would just give you a random name for a temp file, won't create it. If we were to create the file ourselves, we could use `open` or a subprocess to let the shell write to it. Between these two steps, an attacker can (hypothetically) create a file with such name and modify the contents.

But, in case of `mkstemp`, the method creates the file for you, with a strict permission of 600. Meaning no other user except the user will be able to modify it. What if an attacker creates a file with that name at first? In that case, `mkstemp` will look if there's an existing file with that name. If there is, it'll create another random name to suggest and so on.

Which is why, even though the temp file is not being created by the Python program, we should use `mkstemp`- as long as it's being used as a "file". However, if `mktemp` is used as a random name generator (where no file operation are required), in those cases we can ignore such fixes.

Reply:
Ok, I get it, my bad, makes sense. Going to merge this one as well.